aspnetcore/security/authorization/secure-data.md

```diff
@@ -0,0 +1,314 @@
+| test@example.com | Can edit/delete own data |
+
+Create a contact in the administrators browser. Copy the URL for delete and edit from the administrator contact. Paste these links into the test users browser to verify the test user cannot perform these operations.
+
+## Create the starter app
+
+Follow these instructions to create the starter app.
+
+* Create a new **ASP.NET Core Web Application** using [Visual Studio 2017](https://www.visualstudio.com/en-us/visual-studio-homepage-vs.aspx) named "ContactManager"
+
+  * Create the app with **Individual User Accounts**.
+  * Name it "ContactManager" so your namespace will match the namespace use in the sample.
+
+* Add the following `Contact` model:
+
+  [!code-csharp[Main](secure-data/samples/starter/Models/Contact.cs?name=snippet1)]
+
+* Scaffold the `Contact` model using Entity Framework Core and the `ApplicationDbContext` data context. Accept all the scaffolding defaults. Using `ApplicationDbContext` for the data context class  puts the contact table in the [Identity](xref:security/authentication/identity) database. See [Adding a model](xref:tutorials/first-mvc-app/adding-model) for more information.
+
+* Update the **ContactManager** anchor in the *Views/Shared/_Layout.cshtml* file from `asp-controller="Home"` to `asp-controller="Contacts"` so tapping the **ContactManager** link will invoke the Contacts controller. The original markup:
+
+```html
+   <a asp-area="" asp-controller="Home" asp-action="Index" class="navbar-brand">ContactManager</a>
+   ```
+
+The updated markup:
+
+```html
+   <a asp-area="" asp-controller="Contacts" asp-action="Index" class="navbar-brand">ContactManager</a>
+   ```
+
+* Scaffold the initial migration and update the database
+
+```none
+   dotnet ef migrations add initial
+   dotnet ef database update
+   ```
+
+* Test the app by creating, editing and deleting a contact
+
+## Seed the database
+
+Add the `SeedData` class to the *Data* folder. If you've downloaded the sample, you can copy the *SeedData.cs* file to the *Data* folder of the starter project.
+
+[!code-csharp[Main](secure-data/samples/starter/Data/SeedData.cs)]
+
+Add the highlighted code to the end of the `Configure` method in the *Startup.cs* file:
+
+[!code-csharp[Main](secure-data/samples/Starter/Startup.cs?name=Configure&highlight=28-)]
+
+Test that the app seeded the database. The seed method does not run if there are any rows in the contact DB.
+
+### Create the resources for the tutorial
+
+* Create a folder named *Authorization*.
+* Copy the *Authorization\ContactOperations.cs* file from the completed project download, or copy the following code:
+
+[!code-csharp[Main](secure-data/samples/final/Authorization/ContactOperations.cs)]
+
+<a name=secure-data-add-resources-label></a>
+
+### Additional resources
+
+* [ASP.NET Core Authorization Lab](https://github.com/blowdart/AspNetAuthorizationWorkshop)
+* [Authorization](index.md)
```

Comment:
Authorization --> Authorization in ASP.NET Core : Simple, role, claims-based and custom